### ISSUE TITLE
[Bug] dubbo RequestBody list Deserialization failure

### ISSUE THREAD

**Issue by smalltalkhyb** [apache/dubbo#14466]
### Pre-check

- [X] I am sure that all the content I provide is in English.

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dubbo/issues?q=is%3Aissue) and found no similar issues.

### Apache Dubbo Component

Java SDK (apache/dubbo)

### Dubbo Version

JDK 17    <dubbo3.version>3.3.0-beta.4</dubbo3.version>

### scene
we use triple  protocol  

send  request like  :

`curl --location 'http://127.0.0.1:8080/oss/batchSyncFileInfo' \
--header 'x-mse-tag: xingzhihe' \
--header 'User-Agent: Apifox/1.0.0 (https://apifox.com)' \
--header 'Content-Type: application/json' \
--data '[{
        "platform": 2,
        "fileName": "ocr-1721699518782.png",
        "directoryId": "62871482305351680",
        "uniqueFileName": "E0FC312A-C065-4F5A-86CB-85DBA6A9AC26-1721699520039_.ocr-1721699518782.png",
        "ossFileKey": "temp/E0FC312A-C065-4F5A-86CB-85DBA6A9AC26-1721699520039_.ocr-1721699518782.png",
        "fileSize": 3
}]'`

but result error:

org.springframework.core.convert.ConversionFailedException: Failed to convert from type [java.util.ArrayList<?>] to type [@org.springframework.web.bind.annotation.RequestBody @jakarta.validation.Valid java.util.List<cn.itbox.uap.sys.setting.application.model.OssSyncFileInfoRequest>] for value '[{platform=2, fileName=ocr-1721699518782.png, directoryId=62871482305351680, uniqueFileName=E0FC312A-C065-4F5A-86CB-85DBA6A9AC26-1721699520039_.ocr-1721699518782.png, ossFileKey=temp/E0FC312A-C065-4F5A-86CB-85DBA6A9AC26-1721699520039_.ocr-1721699518782.png, fileSize=3}]'

### code

```

 @PostMapping(value = "/oss/batchSyncFileInfo")
    List<OssSyncFileInfoResponse> batchSyncFileInfo(@RequestBody @Valid List<OssSyncFileInfoRequest> list) throws IOException;


    @Override
    public List<OssSyncFileInfoResponse> batchSyncFileInfo(@RequestBody @Valid List<OssSyncFileInfoRequest> list) throws IOException {
        List<OssSyncFileInfoResponse> result = new ArrayList<>();
        for (OssSyncFileInfoRequest fileInfo : list) {
            result.add(this.syncFileInfo(fileInfo));
        }
        return result;
    }
```

OssSyncFileInfoRequest  is defined 

```
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class OssSyncFileInfoRequest implements Serializable {

    /**
     * 原始文件名
     */
    @NotBlank(message = "fileName 不能为空")
    private String fileName;
    /**
     * 文件唯一名
     */
    @NotBlank(message = "uniqueFileName 不能为空")
    private String uniqueFileName;

    /**
     * oss的文件唯一标识
     */
    @NotBlank(message = "ossFileKey 不能为空")
    private String ossFileKey;

    /**
     * 从基础平台获取的目录id
     */
    @NotNull(message = "directoryId 不能为空")
    private Long directoryId;

    /**
     * 文件大小
     */
    @NotNull(message = "fileSize 不能为空")
    @Positive(message = "fileSize 超出范围")
    private Long fileSize;

    /**
     * 哪个平台上传的 1.B端，2ios，3安卓
     */
    @Range(min = 1, max = 3, message = "platform 超出范围")
    private Integer platform;


    /**
     * 是否解压
     */
    private Boolean shouldUnzip = false;

    /**
     * 文件的md5值
     */
    private String md5;

}`

```

### return result

···
{
    "code": "0000",
    "message": "suceess",
    "data": null,
    "success": true,
    "timestamp": 1721720506935
}
···

### debug info 
![image](https://github.com/user-attachments/assets/c9ea531b-e55f-4f4d-915d-08cbb5814864)

![image](https://github.com/user-attachments/assets/bf51e249-e3bc-4f2c-8e2c-8411971957c5)

![image](https://github.com/user-attachments/assets/a410561f-39f8-43ed-9308-4afb9e1d4ea6)

![image](https://github.com/user-attachments/assets/675953f6-a711-43b8-ac6a-d74a862a21d9)

### Correct code
we repair and expend this ,now is run ok 

![image](https://github.com/user-attachments/assets/624df978-f39f-40a0-8e64-8817fd2ccb3a)

### Are you willing to submit a pull request to fix on your own?

- [X] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by oxsean** [comment]
Thanks for report, I will take a look.

**Comment by oxsean** [comment]
@smalltalkhyb fixed in latest 3.3 branch, could you please build and test again?
